File: Code_Help_By_Babbar/23_2D_arrays/03_row_and_col_wise_sum.cpp
#include <iostream>
using namespace std;

int main()
{

    int arr[3][4] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};

    cout << "Printing The 2D Array" << endl;
    for (int row = 0; row < 3; row++)
    {
        for (int col = 0; col < 4; col++)
        {
            cout << arr[row][col] << " ";
        }
        cout << endl;
    }

    cout << endl;
    cout << "Printing row wish sum " << endl;
    for (int row = 0; row < 3; row++)
    {
        int sum = 0;
        for (int col = 0; col < 4; col++)
        {
            sum = sum + arr[row][col];
        }
        cout << sum << " ";
        cout << endl;
    }

    cout << endl;
    cout << endl;
    cout << "Printing column wish sum " << endl;
    for (int col = 0; col < 4; col++)
    {
        int sum = 0;
        for (int row = 0; row < 3; row++)
        {
            sum = sum + arr[row][col];
        }
        cout << sum << " ";
    }

    return 0;
}

File: Topics_to_Focus_On_JS/[ 01 ] Basic Data Structure/01_Array/03_Matrix/03_row_and_col_wise_sum.js
function twoDMatrix(arr) {
  const row = 3;
  const col = 3;
  const t = Math.floor(Math.sqrt(arr.length));
  const matrix = Array.from({ length: t }, () => new Array(t).fill(0));
  let count = 0;

  // print row wise
  for (let m = 0; m < row; m++) {
    for (let n = 0; n < col; n++) {
      matrix[m][n] = arr[count];
      count++;
    }
  }

  // row wise sum
  for (let m = 0; m < row; m++) {
    let sum = 0;
    for (let n = 0; n < col; n++) {
      sum = sum + matrix[m][n];
    }
    console.log(sum);
  }

  // row wise sum
  for (let n = 0; n < col; n++) {
    let sum = 0;
    for (let m = 0; m < row; m++) {
      sum = sum + matrix[m][n];
    }
    console.log(sum);
  }

  return matrix;
}

const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const res = twoDMatrix(arr);
console.log(res);
